### PR TITLE
Run ESLint on internal scripts

### PR DESCRIPTION
**Pull request by basil**
`.eslintignore` was ignoring `src/main/webapp/scripts/*.js` with the comment "External scripts", which was incorrect: some of those are external scripts, but others are internal scripts. They should not be ignored, as linting is a valuable service. This PR unignores these scripts, fixes a few ESLint violations that were easy, any ignores other ESLint violations that were difficult. This will at least prevent new ESLint violations from being introduced in the future.

### Testing done

`mvn clean verify -DskipTests`

### Proposed changelog entries

N/A

### Proposed upgrade guidelines

N/A

### Submitter checklist

- [ ] The Jira issue, if it exists, is well-described.
- [ ] The changelog entries and upgrade guidelines are appropriate for the audience affected by the change (users or developers, depending on the change) and are in the imperative mood (see [examples](https://github.com/jenkins-infra/jenkins.io/blob/master/content/_data/changelogs/weekly.yml)).
  - Fill in the **Proposed upgrade guidelines** section only if there are breaking changes or changes that may require extra steps from users during upgrade.
- [ ] There is automated testing or an explanation as to why this change has no tests.
- [ ] New public classes, fields, and methods are annotated with `@Restricted` or have `@since TODO` Javadocs, as appropriate.
- [ ] New deprecations are annotated with `@Deprecated(since = "TODO")` or `@Deprecated(forRemoval = true, since = "TODO")`, if applicable.
- [ ] New or substantially changed JavaScript is not defined inline and does not call `eval` to ease future introduction of Content Security Policy (CSP) directives (see [documentation](https://www.jenkins.io/doc/developer/security/csp/)).
- [ ] For dependency updates, there are links to external changelogs and, if possible, full differentials.
- [ ] For new APIs and extension points, there is a link to at least one consumer.

### Desired reviewers

@mention

### Maintainer checklist

Before the changes are marked as `ready-for-merge`:

- [ ] There are at least two (2) approvals for the pull request and no outstanding requests for change.
- [ ] Conversations in the pull request are over, or it is explicit that a reviewer is not blocking the change.
- [ ] Changelog entries in the pull request title and/or **Proposed changelog entries** are accurate, human-readable, and in the imperative mood.
- [ ] Proper changelog labels are set so that the changelog can be generated automatically.
- [ ] If the change needs additional upgrade steps from users, the `upgrade-guide-needed` label is set and there is a **Proposed upgrade guidelines** section in the pull request title (see [example](https://github.com/jenkinsci/jenkins/pull/4387)).
- [ ] If it would make sense to backport the change to LTS, a Jira issue must exist, be a _Bug_ or _Improvement_, and be labeled as `lts-candidate` to be considered (see [query](https://issues.jenkins.io/issues/?filter=12146)).

<a href="https://gitpod.io/#https://github.com/jenkinsci/jenkins/pull/8012"></a>

### CODE REVIEW

File: war/.eslintignore
Comment (basil) [inline war/.eslintignore:17]: Cleaning up after #6908 which removed `jest.config.js` but forgot to remove this exclusion.

File: war/.eslintrc.js
Comment (basil) [inline war/.eslintrc.js:80]: Cleaning up after #6908 which removed `jest.config.js` but forgot to remove these exclusions.

File: war/src/main/webapp/scripts/hudson-behavior.js
Comment (timja) [inline war/src/main/webapp/scripts/hudson-behavior.js:2289]: according to the docs it should be marked as exported rather than disabling the rule, it would end up with the same result though so non-blocking
```suggestion
/* exported toQueryString */
```

**Review by timja — APPROVED** [review]

File: war/src/main/webapp/scripts/hudson-behavior.js
Comment (basil) [reply war/src/main/webapp/scripts/hudson-behavior.js:2294]: This suggestion causes the build to go from passing to failing with

```
  2295:10  error  'toQueryString' is defined but never used  no-unused-vars
```

thereby making it a non-suggestion.

Comment (timja) [reply war/src/main/webapp/scripts/hudson-behavior.js:2294]: right it's because some of the files are modules. Which that feature doesn't work if modules are enabled.

to make it work there would need to be another eslintrc file for the non modular code I think which is likely not worth it.

**Comment by timja** [comment]
/label ready-for-merge

----

This PR is now ready for merge, after ~24 hours, we will merge it if there's no negative feedback.

Thanks!

**Pull request merged**
